src/validation.cpp

```diff
@@ -5034,6 +5034,13 @@ void ChainstateManager::LoadExternalBlockFile(
     LogPrintf("Loaded %i blocks from external file in %dms\n", nLoaded, Ticks<std::chrono::milliseconds>(SteadyClock::now() - start));
 }
 
+bool ChainstateManager::ShouldCheckBlockIndex() const
+{
+    if (!*Assert(m_options.check_block_index)) return false;
```

Comment:
In commit "validation: allow to specify frequency for -checkblockindex" (b51eb422d657bb1d0e6f9c7e71eb8a87a8c74f42)

Note: I was initially confused why this is has an Assert to check if the option is set, instead of just returning a default value when the option is unset, but the reason seems to be that this is a check to make sure that Flatten() has been called and `opts.chainparams.DefaultConsistencyChecks()` has been applied. Could maybe add a comment above this like "// Assert to verify Flatten() has been called." to clarify this

Reply:
I added a comment. By the way, the Assert is also already present on master.